bootstrap/bootstrapper/fs/source.go

```diff
@@ -88,18 +71,25 @@ func (fss *fileSystemSource) GetAvailability(shard uint32, targetRangesForShard
 
 // ReadData returns raw series for a given shard within certain time ranges.
 func (fss *fileSystemSource) ReadData(shard uint32, tr xtime.Ranges) (m3db.ShardResult, xtime.Ranges) {
-	log := fss.opts.GetLogger()
-
 	if xtime.IsEmpty(tr) {
 		return nil, tr
 	}
-	files, err := fss.getInfoFiles(shard)
-	if err != nil {
-		log.Errorf("unable to get info files for shard %d: %v", shard, err)
+
+	var files []string
+	fs.ForEachInfoFile(
```

Comment:
Yeah so this is because I'd like to compare the checksum of the info data against the when determining which info files are valid, and that requires reading the info files, otherwise we'll need to delay validation till below when we actually read the files, which is more heavyweight than just reading the info files.